### PR TITLE
Big overhaul for 5.0

### LINKED ISSUE #41: UseSnakeCaseNamingConvention() Does not Handle Owned Entities Table in 5.0.0.rc
For example, for a context like this

```

namespace Domain.Entities.Accounts
{
    public class Account : IEntity<int> 
    {
        public int Id { get; set; }
        public string AccountNumber { get; set; } = null!;
     

        public AccountBalance AccountBalance { get; set; } = null!;   //Owned Type
      
    }


}
 public class AccountBalance //: IEntity<int>
   {    
       
     public Decimal? AuthorizedBalance { get; set; }
       

  }
	
 public void Configure(EntityTypeBuilder<Account> builder)
        {
            builder.ToTable("account",schema: "ops");
            builder.HasKey(a => a.Id);
          
            
            builder.OwnsOne(a => a.AccountBalance, ownedBuilder =>
            {
               
                ownedBuilder.WithOwner().HasForeignKey("AccountId");
                ownedBuilder.Property("AccountId").HasColumnName("account_id");
                ownedBuilder.ToTable("account_balance", schema: "ops");
               
            });

      }
```

When run  issue still persist in 5.0.0-rc.  After i run Update-Database -Context CBSContext

ther following error shown

```
Npgsql.PostgresException (0x80004005): 42P07: relation "pk_accounts" already exists
   at Npgsql.NpgsqlConnector.<>c__DisplayClass160_0.<<DoReadMessage>g__ReadMessageLong|0>d.MoveNext()
--- End of stack trace from previous location ---
   at Npgsql.NpgsqlConnector.<>c__DisplayClass160_0.<<DoReadMessage>g__ReadMessageLong|0>d.MoveNext()
--- End of stack trace from previous location ---
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming)
   at Npgsql.NpgsqlDataReader.NextResult()
   at Npgsql.NpgsqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteNonQuery(Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteNonQuery()
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteNonQuery(RelationalCommandParameterObject parameterObject)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationCommand.ExecuteNonQuery(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Migrations.Internal.MigrationCommandExecutor.ExecuteNonQuery(IEnumerable`1 migrationCommands, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.Migrate(String targetMigration)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations.UpdateDatabase(String targetMigration, String connectionString, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.UpdateDatabaseImpl(String targetMigration, String connectionString, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.UpdateDatabase.<>c__DisplayClass0_0.<.ctor>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
  Exception data:
    Severity: ERROR
    SqlState: 42P07
    MessageText: relation "pk_accounts" already exists
    File: d:\pginstaller_13.auto\postgres.windows-x64\src\backend\catalog\index.c
    Line: 866
    Routine: index_create
42P07: relation "pk_accounts" already exists
```

If i remove options.UseSnakeCaseNamingConvention(); migration works fine ..

### LINKED ISSUE #45: Implement TPH properly

### LINKED ISSUE #46: Implement support for TPT

### PR DESCRIPTION
**Pull request by roji**
* Use EF Core 5.0
* Fixes around TPH, owned entity management
* Redid tests

Fixes #45
Closes #46
Fixes #41

@AndriySvyryd if you feel like it, you can take a look at this. I know there are various edge cases which aren't taken care of (conventions are complicated), but hopefully this covers most mainstream scenarios, including new stuff in 5.0.

### CODE REVIEW

**Comment by CumpsD** [comment]
Looking forward to 5.0.0 final

File: EFCore.NamingConventions/Internal/NameRewritingConvention.cs [hunk 1]
```diff
@@ -21,17 +26,60 @@ public class NameRewritingConvention :
...
-            => propertyBuilder.HasColumnName(_namingNameRewriter.RewriteName(propertyBuilder.Metadata.GetColumnName()));
+            IConventionPropertyBuilder propertyBuilder,
+            IConventionContext<IConventionPropertyBuilder> context)
+        {
+            var entityType = propertyBuilder.Metadata.DeclaringEntityType;
+            var property = propertyBuilder.Metadata;
+
+            property.SetColumnName(_namingNameRewriter.RewriteName(property.GetColumnBaseName()));
 
+            foreach (var storeObjectType in _storeObjectTypes)
```
Comment (AndriySvyryd) [inline EFCore.NamingConventions/Internal/NameRewritingConvention.cs:70]: Why do you rewrite column names for all identifier types, but only rewrite identifier name for tables?

File: EFCore.NamingConventions/Internal/NameRewritingConvention.cs [hunk 2]
```diff
@@ -61,7 +127,7 @@ public void ProcessForeignKeyOwnershipChanged(IConventionForeignKeyBuilder relat
                     .Except(ownedEntityType.FindPrimaryKey().Properties)
                     .Where(p => p.Builder.CanSetColumnName(null)))
                 {
-                    var columnName = property.GetColumnName();
+                    var columnName = property.GetColumnBaseName();
                     var prefix = _namingNameRewriter.RewriteName(ownedEntityType.ShortName());
                     if (!columnName.StartsWith(prefix, StringComparison.OrdinalIgnoreCase))
```
Comment (AndriySvyryd) [inline EFCore.NamingConventions/Internal/NameRewritingConvention.cs:132]: Might make sense to also check for the non-rewritten prefix

File: EFCore.NamingConventions/Internal/NameRewritingConvention.cs (hunk 1 above)
Comment (roji) [reply EFCore.NamingConventions/Internal/NameRewritingConvention.cs:70]: The logic here was that while table names are set by convention, view mapping is done by calling ToView which requires the view name to be set explicitly - is my mental model wrong? I understand that there can still be a convention (from a plugin, provider) which maps to views (is this what you had in mind?), so I've just added rewriting view names too. 

For DbFunction I guess the logic is the same - any situation apart from a plugin or provider convention where one would be created by convention?

File: EFCore.NamingConventions/Internal/NameRewritingConvention.cs (hunk 2 above)
Comment (roji) [reply EFCore.NamingConventions/Internal/NameRewritingConvention.cs:132]: Thanks, added that.
